File: 129-sum-root-to-leaf-numbers/129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void rootToleaf(TreeNode* root, string currStr, int& ans){
        if(!root->left && !root->right){
            currStr += to_string(root->val);
            ans += stoi(currStr,0);
            return;
        }
        string curr = to_string(root->val);
        if(root->left!=NULL) rootToleaf(root->left,currStr+curr,ans);
        if(root->right!=NULL) rootToleaf(root->right,currStr+curr,ans);
    }
    
    int sumNumbers(TreeNode* root) {
        int ans = 0;
        rootToleaf(root,"",ans);
        return ans;
    }
};

File: 129-sum-root-to-leaf-numbers/129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    ans = 0
    def rootToleaf(self,root,s):
        if(not root.left and not root.right):
            s += str(root.val)
            self.ans += int(s)
            return
        curr = str(root.val)
        if(root.left!=None): self.rootToleaf(root.left,s+curr)
        if(root.right!=None): self.rootToleaf(root.right,s+curr)
    
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        self.ans = 0
        self.rootToleaf(root,"")
        return self.ans